Framework/Geometry/src/Instrument/DetectorInfo.cpp

```diff
@@ -158,7 +159,13 @@ double DetectorInfo::twoTheta(const size_t index) const {
   }
 
   const auto sampleDetVec = position(index) - samplePos;
-  return sampleDetVec.angle(beamLine);
+  try {
```

Comment:
I'm going to have to hold this up for the moment I'm afraid as I think this is too low-level a place to do this. As `0.0` could be a real angle then from the outside it's impossible for a user of this method to tell if `0.0` means it really is 0.0 or an error occurred. As this is low level then some other user of the method may want to do to do something different if the exceptional case happens.

Can I ask if we know why the POLARIS IDF has these detectors and is it just `ConvertUnits` that fails?